src/Sentry/Platforms/Native/UnmanagedHttpContent.cs

```diff
@@ -0,0 +1,63 @@
+using Sentry.Extensibility;
+
+namespace Sentry.Native;
+
+internal class UnmanagedHttpContent : SerializableHttpContent
+{
+    private readonly IntPtr _content;
+    private readonly int _length = 0;
+    private readonly IDiagnosticLogger? _logger;
+
+    public UnmanagedHttpContent(IntPtr content, int length, IDiagnosticLogger? logger)
+    {
+        _content = content;
+        _length = length;
+        _logger = logger;
+    }
+
+    protected override Task SerializeToStreamAsync(Stream stream, TransportContext? context)
+    {
+        try
+        {
+            unsafe
+            {
+                using var unmanagedStream = new UnmanagedMemoryStream((byte*)_content.ToPointer(), _length);
+                return unmanagedStream.CopyToAsync(stream);
```

Comment:
attention: `CopyToAsync` is not `await`'d

Is this intended?
Should this method be `async`?

Reply:
It is intended because of `unsafe`:

> Cannot await in an unsafe context [CS4004](https://msdn.microsoft.com/query/roslyn.query?appId=roslyn&k=k(CS4004))

Reply:
TIL ... right ... because of lifting local variables to fields of the compiler-generated StateMachine